Src/Sarif.PatternMatcher/AnalyzeCommand.cs

```diff
@@ -122,36 +122,79 @@ internal static void PushInheritedData(SearchDefinitions definitions, Dictionary
+                                                             definition.SharedStrings,
+                                                             sharedStrings);
+
                 matchExpression.FileNameDenyRegex ??= definition.FileNameDenyRegex;
 
-                matchExpression.FileNameAllowRegex = PushData(matchExpression.FileNameAllowRegex, sharedStrings);
+                matchExpression.FileNameAllowRegex = PushData(matchExpression.FileNameAllowRegex,
+                                                             definition.SharedStrings,
+                                                             sharedStrings);
+
                 matchExpression.FileNameAllowRegex ??= definition.FileNameAllowRegex;
 
-                matchExpression.ContentsRegex = PushData(matchExpression.ContentsRegex, sharedStrings);
+                matchExpression.ContentsRegex = PushData(matchExpression.ContentsRegex,
+                                                         definition.SharedStrings,
+                                                         sharedStrings);
+
+                if (matchExpression.Level == 0)
+                {
+                    matchExpression.Level = definition.Level;
+                }
+            }
+        }
+
+        private static string PushData(string text, params Dictionary<string, string>[] sharedStringsDictionaries)
+        {
+            if (text?.Contains("$") != true)
+            {
+                return text;
+            }
+
+            foreach (Dictionary<string, string> sharedStrings in sharedStringsDictionaries)
+            {
+                if (sharedStrings == null)
+                {
+                    continue;
+                }
 
-                if (matchExpression.Level == 0) { matchExpression.Level = definition.Level; }
+                foreach (string key in sharedStrings.Keys)
+                {
+                    text = text.Replace(key, sharedStrings[key]);
+                }
             }
+
+            return text;
         }
 
         internal static Dictionary<string, string> LoadSharedStrings(string sharedStringsFullPath, IFileSystem fileSystem)
         {
             var result = new Dictionary<string, string>();
 
-            foreach (string line in fileSystem.FileReadAllLines(sharedStringsFullPath))
+            foreach (string fileLine in fileSystem.FileReadAllLines(sharedStringsFullPath))
             {
+                string line = fileLine.Trim();
+                if (string.IsNullOrEmpty(line) || line.StartsWith("#")) { continue; }
+
                 int index = line.IndexOf('=');
                 if (index == -1) { ThrowInvalidSharedStringsEntry(line); }
 
                 string key = line.Substring(0, index);
                 if (!key.StartsWith("$")) { ThrowInvalidSharedStringsEntry(line); }
 
-                result[key] = line.Substring(key.Length + "=".Length);
+                string value = line.Substring(key.Length + "=".Length);
```

Comment:
>string value [](start = 16, length = 12)

do we need to do this?

Reply:
do what? declare a short-lived variable?

---
In reply to: [555679637](https://github.com/microsoft/sarif-pattern-matcher/pull/101#discussion_r555679637) [](ancestors = 555679637)